src/route.shared.ts

```diff
@@ -82,131 +71,52 @@ function getErrorPageId(allPageIds: string[]): string | null {
-  const matchValue = routeString.length
-  const routeParams = match.params
-  return { matchValue, routeParams }
-}
+function compileRouteFunctionsForUrl(routes: PageRoute[], url: string, contextProps: Record<string, unknown>): CompiledRouteFunction[] {
+  const { customRouting: { sortRoutes=defaultSortRoutes }={} } = getSsrEnv();
+  const functionalRoutes : RouteFunction[] = (routes as RouteFunction[])
+    .filter(route => isCallable(route.pageRoute))
 
-function isStaticRoute(route: string): boolean {
-  const { matchValue, routeParams } = routeWith_pathToRegexp(route, route)
-  return matchValue !== false && Object.keys(routeParams).length === 0
-}
+  const compiledRoutes : CompiledRouteFunction[] = functionalRoutes
+    .map(route => {
+      assert(route.pageRouteFile);
+      const routeFunctionResult : (string|RouteFunctionMatch) = resolveRouteFunction(route.pageRoute, url, contextProps, route.pageRouteFile);
+      
+      return { ...route, pageRoute: routeFunctionResult };
+    });
 
-function routeWith_filesystem(urlPathname: string, pageId: string, allPageIds: PageId[]): boolean {
-  const pageRoute = getFilesystemRoute(pageId, allPageIds)
-  urlPathname = normalizeUrl(urlPathname)
-  // console.log('[Route Candidate] url:' + urlPathname, 'pageRoute:' + pageRoute)
-  const matchValue = urlPathname === pageRoute
-  return matchValue
-}
-function normalizeUrl(urlPathname: string): string {
-  return '/' + urlPathname.split('/').filter(Boolean).join('/').toLowerCase()
-}
-function getFilesystemRoute(pageId: string, allPageIds: string[]): string {
-  let pageRoute = removeCommonPrefix(pageId, allPageIds)
-  pageRoute = pageRoute
-    .split('/')
-    .filter((part) => part !== 'index')
-    .join('/')
-  pageRoute = normalizeUrl(pageRoute)
-  return pageRoute
-}
-function removeCommonPrefix(pageId: PageId, allPageIds: PageId[]) {
-  const relevantPageIds = allPageIds.filter((pageId) => !isErrorPage(pageId))
-  const commonPrefix = getCommonPath(relevantPageIds)
-  assert(pageId.startsWith(commonPrefix))
-  return pageId.slice(commonPrefix.length)
-}
-function getCommonPath(pageIds: string[]): string {
-  pageIds.forEach((pageId) => {
-    assertUsage(
-      !pageId.includes('\\'),
-      'Your directory names and file names are not allowed to contain the character `\\`'
-    )
-  })
-  const pageIdList = pageIds.concat().sort()
-  const first = pageIdList[0]
-  const last = pageIdList[pageIdList.length - 1]
-  let idx = 0
-  for (; idx < first.length; idx++) {
-    if (first[idx] !== last[idx]) break
-  }
-  const commonPrefix = first.slice(0, idx)
-  const pathsPart = commonPrefix.split('/')
-  assert(pathsPart.length >= 2)
-  const commonPath = slice(pathsPart, 0, -1).join('/') + '/'
-  return commonPath
+  return compiledRoutes
+    .sort(sortRoutes)
 }
 
-/**
-  Returns the ID of all pages including `_error.page.*` but excluding `_default.page.*`.
-*/
-async function getPageIds(): Promise<PageId[]> {
```

Comment:
Did something chagne here? Can we keep it the function in the file for now? (Because right now I've to compare each line to see if there is a diff.)